Framework/API/src/ExperimentInfo.cpp

```diff
@@ -1004,15 +1004,18 @@ ExperimentInfo::getInstrumentFilename(const std::string &instrumentName,
     // find the first beat file
     for (Poco::DirectoryIterator dir_itr(directoryName); dir_itr != end_iter;
          ++dir_itr) {
-      if (!Poco::File(dir_itr->path()).isFile())
+
+      const auto filePath = dir_itr.path();
+      if (!filePath.isFile())
         continue;
 
-      std::string l_filenamePart = Poco::Path(dir_itr->path()).getFileName();
+      std::string l_filenamePart = filePath.getFileName();
       if (regex_match(l_filenamePart, regex)) {
-        g_log.debug() << "Found file: '" << dir_itr->path() << "'\n";
+        const auto pathName = dir_itr->path();
```

Comment:
Could we avoid the confusion with the `*` & `->` operators on `DirectoryIterator`? As we have the file now this should be the same as `filePath.path()`. Also a `const auto &` here would avoid a copy too.